File: go/clock/clock.go
package clock

import (
	"fmt"
)

const testVersion = 4

const MinutesOfDay = 1440

type Clock struct {
	hour   int
	minute int
}

func NewWithMinutes(minutes int) Clock {
	mm := minutes % MinutesOfDay
	if mm < 0 {
		mm = mm + MinutesOfDay
	}
	return Clock{
		hour:   mm / 60,
		minute: mm % 60,
	}
}

func New(h, m int) Clock {
	return NewWithMinutes(h*60 + m)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

func (c Clock) Add(minutes int) Clock {
	return NewWithMinutes(c.hour*60 + c.minute + minutes)
}

File: javascript/clock/clock.js
const MINUTES_PER_DAY = 60 * 24

const pad = val => {
  const t = String(val)
  return t.length === 1 ? `0${t}` : t
}

const turnClockHands = minutes => {
  const ms = minutes % MINUTES_PER_DAY + (minutes < 0 ? MINUTES_PER_DAY : 0)
  const minute = ms % 60
  const hour = (ms - minute) / 60
  return { hour, minute }
}

class Clock {
  constructor(minutes) {
    const { hour, minute } = turnClockHands(minutes)
    this._hour = hour
    this._minute = minute
  }

  get hour() {
    return this._hour
  }

  get minute() {
    return this._minute
  }

  toMinutes() {
    return this.hour * 60 + this.minute
  }

  plus(added) {
    const minutes = this.toMinutes() + added
    return new Clock(minutes)
  }

  minus(minutes) {
    return this.plus(-minutes)
  }

  toString() {
    return `${pad(this.hour)}:${pad(this.minute)}`
  }

  equals(clock) {
    if (clock instanceof Clock) {
      return this.hour === clock.hour && this.minute === clock.minute
    }
    return false
  }
}

const at = (hour, minute = 0) => {
  const minutes = hour * 60 + minute
  return new Clock(minutes)
}

module.exports = { at }
